packages/discord.js/src/structures/InteractionCollector.js

```diff
@@ -147,10 +147,17 @@ class InteractionCollector extends Collector {
      * @event InteractionCollector#collect
      * @param {BaseInteraction} interaction The interaction that was collected
      */
+
     if (this.interactionType && interaction.type !== this.interactionType) return null;
     if (this.componentType && interaction.componentType !== this.componentType) return null;
     if (this.messageId && interaction.message?.id !== this.messageId) return null;
-    if (this.messageInteractionId && interaction.message?.interaction?.id !== this.messageInteractionId) return null;
+    if (
+      this.messageInteractionId &&
+      interaction.message?.interaction?.id &&
+      interaction.message?.interaction?.id !== this.messageInteractionId
```

Comment:
Previous line makes sure none of those objects are nullish.
```suggestion
      interaction.message.interaction.id !== this.messageInteractionId
```

Reply:
Prior to this PR it was the way you're requesting it to be, however I was running into a `null !== undefined` edgecase that was causing it to break.